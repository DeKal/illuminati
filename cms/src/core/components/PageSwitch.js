import React from 'react'
import { Switch, Route } from 'react-router-dom'
import Container from 'core/components/page/Container'
import Dashboard from 'dashboard/components/Dashboard'
import User from 'user/components/User'
import { pages } from 'core/const/pages'

const PageSwitch = () => (
  <Container>
    <Switch>
      <Route exact path={pages.dashboard.url}>
        <Dashboard />
      </Route>
      <Route path={pages.profile.url}>
        <User />
      </Route>
    </Switch>
  </Container>
)

export default PageSwitch
