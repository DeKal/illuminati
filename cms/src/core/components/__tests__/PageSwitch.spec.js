import * as React from 'react'
import { shallow } from 'enzyme'
import PageSwitch from '../PageSwitch'

describe('PageSwitch', () => {
  it('should render PageSwitch with no error', function() {
    const wrap = shallow(<PageSwitch />)
    expect(wrap).toMatchSnapshot()
  })
})
