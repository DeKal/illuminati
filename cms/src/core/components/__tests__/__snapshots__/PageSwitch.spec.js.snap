// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageSwitch should render PageSwitch with no error 1`] = `
<Styled(div)>
  <Switch>
    <Route
      exact={true}
      path="/"
    >
      <DashBoard />
    </Route>
    <Route
      path="/profile"
    >
      <User />
    </Route>
    <Route
      path="/posts"
    >
      <Connect(PostsWithHook) />
    </Route>
    <Route
      path="/post/"
    >
      <Post />
    </Route>
  </Switch>
</Styled(div)>
`;
